### ISSUE TITLE
fluent-bit link error with -DFLB_PROXY_GO=0

### ISSUE THREAD

**Issue by kkm000** [fluent/fluent-bit#1440]
This is a regression from the 1.1 branch present in both 1.2 and master.

```
[100%] Linking C executable ../bin/fluent-bit
cd /home/kkm/work/fluent-bit/build/src && /usr/bin/cmake -E cmake_link_script CMakeFiles/fluent-bit-bin.dir/link.txt --verbose=1
/usr/bin/gcc-8 -g -O2 -fdebug-prefix-map=/home/kkm/work/fluent-bit=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2
[... SNIP ...]
./src/flb_plugin.c:281: error: undefined reference to 'flb_plugin_proxy_create'
```

`flb_plugin_proxy_create` is defined in the file src/flb_plugin_proxy.c
https://github.com/fluent/fluent-bit/blob/209e8a5ba21e4f0377f11cbe935863ecc6b2aa17/src/flb_plugin_proxy.c#L188-L191

which is compiled conditionally based on `FLB_PROXY_GO`
https://github.com/fluent/fluent-bit/blob/209e8a5ba21e4f0377f11cbe935863ecc6b2aa17/src/CMakeLists.txt#L80-L85

but referenced without any `#ifdef` guard, line 281:
https://github.com/fluent/fluent-bit/blob/209e8a5ba21e4f0377f11cbe935863ecc6b2aa17/src/flb_plugin.c#L265-L282

My CMake-fu is so miserable that I cannot propose a sensible PR, sorry. Probably the missing part is `#ifdef FLB_HAVE_PROXY_GO`, but I do not understand this part of code well (Strictly speaking, I understand _no_ part of fluent-bit code _well_ :-) )
https://github.com/fluent/fluent-bit/blob/209e8a5ba21e4f0377f11cbe935863ecc6b2aa17/CMakeLists.txt#L469-L471

An obvious workaround is compile in the support for `FLB_PROXY_GO=1`, even though I'm not using the Go interface, so this isn't really a biggie. With this setting, link completes successfully.

**Comment by edsiper** [comment]
@kkm000 thanks for reporting and troubleshooting this issue, I've pushed a fix on master and for the next v1.2.2 release: 15f6a9e3

**Comment by kkm000** [comment]
Wow! 16 minutes from report to fix! That's probably the quickest handling of a bug report I seen ever, thank you! :)

**Comment by kkm000** [comment]
Confirm the fix on the tip of the 1.2 branch, thank you again!
